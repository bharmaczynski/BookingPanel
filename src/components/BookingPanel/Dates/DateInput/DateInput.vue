<template>
  <input
      class="date-input font-size-base"
      :value="value"
      :placeholder="placeholder"
      readonly="readonly"
      type="text"
  />
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';

@Component
export default class DateInput extends Vue {
  @Prop({ required: true })
  value!: string

  @Prop({ default: '' })
  placeholder!: string
}
</script>
<style lang="scss" scoped>
.date-input {
  background: transparent;
  border: 0;
  padding: 10px 15px;
  width: 100%;
  cursor: pointer;

  &:focus {
    border: 0;
    outline: 0;
  }
}
</style>
