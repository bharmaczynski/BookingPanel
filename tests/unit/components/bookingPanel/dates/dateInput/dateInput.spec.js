import { shallowMount } from '@vue/test-utils';
import DateInput from '@/components/bookingPanel/dates/dateInput/DateInput.vue';

describe('DateInput', () => {
    let propsData;

    beforeEach(() => {
        propsData = {
            value: '10-10-2021',
        };
    });

    function createWrapper(propsData = {}) {
        return shallowMount(DateInput, {
            propsData,
        });
    }

    describe('Snapshots', () => {
        it('should match initial snapshot', () => {
            const wrapper = createWrapper(propsData);

            expect(wrapper.vm.$el).toMatchSnapshot();
        });

        it('should match snapshot in case placeholder is given', () => {
            propsData = {
                value: '10-10-2021',
                placeholder: 'lorem ipsum',
            };

            const wrapper = createWrapper(propsData);

            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});
