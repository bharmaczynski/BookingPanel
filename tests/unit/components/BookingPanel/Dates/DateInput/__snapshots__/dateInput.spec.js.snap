// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput Snapshots should match initial snapshot 1`] = `
<input
  class="date-input font-size-base"
  placeholder=""
  readonly="readonly"
  type="text"
/>
`;

exports[`DateInput Snapshots should match snapshot in case placeholder is given 1`] = `
<input
  class="date-input font-size-base"
  placeholder="lorem ipsum"
  readonly="readonly"
  type="text"
/>
`;
